src/test/java/org/xstefank/check/TitleCheckTest.java

```diff
@@ -3,22 +3,48 @@
 import org.junit.Assert;
 import org.junit.Test;
 import org.xstefank.TestUtils;
-
+import org.xstefank.model.yaml.RegexDefinition;
 import java.util.regex.Pattern;
 
 public class TitleCheckTest {
 
+    private RegexDefinition titleRegexDefinition = new RegexDefinition();
     private TitleCheck titleCheck;
 
+    @Test (expected=IllegalArgumentException.class)
+    public void testNullTitleParameter() throws IllegalArgumentException {
+        new TitleCheck(null);
+    }
+
+    @Test (expected=IllegalArgumentException.class)
+    public void testNullCommitPatternParameter() throws IllegalArgumentException {
+        titleRegexDefinition.setPattern(null);
+        new TitleCheck(titleRegexDefinition);
+    }
+
     @Test
-    public void checkSimpleRegexMatch() {
-        titleCheck = new TitleCheck(Pattern.compile("^Test.*PR$"));
+    public void testCheckSimpleRegexMatch() {
+        titleRegexDefinition.setPattern(Pattern.compile("^Test.*PR$"));
+        titleCheck = new TitleCheck(titleRegexDefinition);
+
         Assert.assertNull("Cannot match valid regex", titleCheck.check(TestUtils.TEST_PAYLOAD));
     }
 
     @Test
-    public void checkSimpleRegexNonMatch() {
-        titleCheck = new TitleCheck(Pattern.compile("can't.*match.*this"));
+    public void testCheckSimpleRegexNonMatchReturnsExpectedMessage() {
+        titleRegexDefinition.setPattern(Pattern.compile("can't.*match.*this"));
+        titleRegexDefinition.setMessage("This is titleRegexDefinition message");
+        titleCheck = new TitleCheck(titleRegexDefinition);
+
         Assert.assertNotNull("Matched invalid regex", titleCheck.check(TestUtils.TEST_PAYLOAD));
+        Assert.assertEquals("Unexpected message returned", titleRegexDefinition.getMessage(), titleCheck.check(TestUtils.TEST_PAYLOAD));
+    }
+
+    @Test
+    public void testCheckSimpleRegexNonMatchReturnsDefaultMessage() {
+        titleRegexDefinition.setPattern(Pattern.compile("can't.*match.*this"));
+        titleCheck = new TitleCheck(titleRegexDefinition);
```

Comment:
you do not clean titleRegexDefinition instance. if `testCheckSimpleRegexNonMatchReturnsExpectedMessage` runs just before `testCheckSimpleRegexNonMatchReturnsDefaultMessage` will the message be set to DEFAULT_MESSAGE?